modules/utxo-lib/src/bitgo/parseInput.ts

```diff
@@ -289,38 +291,34 @@ function parseP2ms(
   }
 
   return {
-    ...params,
+    scriptType,
     publicKeys: redeemScript.match[':pubkey'],
     pubScript: redeemScript.buffer,
     signatures: match[':signature'] as ParsedSignatureScript2Of3['signatures'],
+    redeemScript: scriptType === 'p2sh' ? redeemScript.buffer : undefined,
+    witnessScript: scriptType === 'p2shP2wsh' || scriptType === 'p2wsh' ? redeemScript.buffer : undefined,
   };
 }
 
 const parseP2sh2Of3: InputParser<ParsedSignatureScript2Of3> = (p) => {
   if (!isLegacy(p)) {
     return new MatchError(`expected legacy input`);
   }
-  return parseP2ms(p.script, {
-    scriptType: 'p2sh',
-  });
+  return parseP2ms(p.script, 'p2sh');
 };
 
 const parseP2shP2wsh2Of3: InputParser<ParsedSignatureScript2Of3> = (p) => {
   if (!isWrappedSegwit(p)) {
     return new MatchError(`expected wrapped segwit input`);
   }
-  return parseP2ms(p.witness, {
-    scriptType: 'p2shP2wsh',
-  });
+  return { ...parseP2ms(p.witness, 'p2shP2wsh'), redeemScript: p.script[0] as Buffer };
```

Comment:
I wonder if this should have a type check on p.script[0] to detect malformed tx deliberately.

Reply:
yup we might want to add this in the future

a malformed script will fail at least at the fullnode so I think it is low-prio